### PR TITLE
feat(vxrn): add basic plugin for removing generateStaticParams from c…

### PR DESCRIPTION
**Pull request by natew**
…lient bundles

### CODE REVIEW

File: packages/vxrn/src/createDevServer.ts [hunk 1]
```diff
@@ -363,7 +363,20 @@ export const createDevServer = async (optionsIn: VXRNConfig) => {
...
-      plugins: [reactNativeHMRPlugin],
+      plugins: [
+        reactNativeHMRPlugin,
+
+        clientBundleTreeShakePlugin({}),
+
+        {
+          name: 'remove-generateStaticParams',
+          transform(code, id, options) {
+            if (code.includes('generateStaticParams')) {
```
Comment (Aslemammad) [inline packages/vxrn/src/createDevServer.ts:374]:
```ts
if (!options.ssr) {
  // use https://github.com/guybedford/es-module-lexer and magic-string to remove the generateStaticParams export
}
```

That's how it's done in the vite world.

File: packages/vxrn/src/plugins/clientBundleTreeShakePlugin.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import type { Node, Program } from 'estree'
+import { walk } from 'estree-walker'
```
Comment (Aslemammad) [inline packages/vxrn/src/plugins/clientBundleTreeShakePlugin.tsx:2]: we can avoid estree-walker and the complexity, es-module-lexer is way simpler and has less things to handle.

File: packages/vxrn/src/plugins/clientBundleTreeShakePlugin.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      }
+    },
+  } satisfies Plugin
+}
+
+function removeImportDeclaration(
+  ast: Program,
+  importName: string,
+  magicString: MagicString
+): boolean {
```
Comment (Aslemammad) [inline packages/vxrn/src/plugins/clientBundleTreeShakePlugin.tsx:77]: Not sure what this function does

Comment (natew) [reply packages/vxrn/src/plugins/clientBundleTreeShakePlugin.tsx:77]: wdym? i did copy this from a nuxt plugin but didnt really check it thoroughly. it removes the imports from that function

Comment (Aslemammad) [reply packages/vxrn/src/plugins/clientBundleTreeShakePlugin.tsx:77]: Ok, I assume it's for removing imports after we remove the params function. If yes, as far as i know, dead code elimination by esbuild/vite would do it automatically, so we might not even need to do it anyway. Not 100% sure though.

**Pull request merged**